### ISSUE TITLE
picard crashes: Could not find the Qt platform plugin "xcb"

### ISSUE THREAD

**Issue by peti** [NixOS/nixpkgs#99458]
Picard is broken in current versions of `master` like 4e71acfe0032d6240fc7c261fb4007e72df82b4a:

~~~~
~ $ picard
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

zsh: abort (core dumped)  picard
~~~~~

According to `git bisect`, the tool was broken by commit 7abb57c7b5eaa9aa34ea289fc1e23ed259547bdb from @jorsn.

**Comment by FlorianFranzen** [comment]
I had similar issues with TeXmacs under X11. As I mostly use Wayland I just ignored the issue so far.

**Comment by jorsn** [comment]
TeXmacs is unrelated. The problem here is that the package uses another qt5 version than pyqt5.
The TeXmacs package uses still qt4, which isn't overridden in `python*Packages`.

**Comment by magnetophon** [comment]
[Carla](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/carla/default.nix) has the same issue.

**Comment by jorsn** [comment]
**I think we need to programmatically find all inconsistent packages.** Almost every package with multiple qt5 versions in its closure is probably broken, and this should not stem from my commit, but root in 3cbcc14eef897607c50be4a9918d88614ea9881e or earlier, as this one set qt-5.14 for all `python*Packages`. `pyqt5` used 5.14 even before. From c88f3adb1793cd3b007baff0292cf1fa8d0ab1e6 until my commit (7abb57c7b5eaa9aa34ea289fc1e23ed259547bdb) some pkgs might not have been broken anymore, because c88f3adb1793cd3b007baff0292cf1fa8d0ab1e6 accidentally reset `python*Packages` to use qt-5.15 again. This, in turn, broke at least `pyqt5_with_qtwebkit`.

**The fix is almost trivial once the broken package is discovered**, since all you need to do is use qt-5.14 for the package. Or you call the package with `python*Packages.callPackage`. The package then inherits the right qt5 version from `python*Packages`. In the package one should use `pythonPackages` in this case, not `python3Packages`. This is because in the python scope (`pythonInterpreters`/`python*Packages`) the former is always set to the current python version, while the latter is inherited from the global scope (`pkgs/top-level/all-packages.nix`).

**Comment by FRidh** [comment]
fix in https://github.com/NixOS/nixpkgs/pull/99956

**Comment by doronbehar** [comment]
#99956 merged.

**Comment by FlorianFranzen** [comment]
> TeXmacs is unrelated. The problem here is that the package uses another qt5 version than pyqt5.
> The TeXmacs package uses still qt4, which isn't overridden in `python*Packages`.

@jorsn Yes, you are right indeed. I forgot #84816 had not been merged yet, so this is probably just an issue for me at the moment. Will update my PR with the fix from #99956 and hope I can find someone with commit right this time around.
